src/main.rs

```diff
@@ -56,9 +58,23 @@ fn main() -> ! {
     // Set up the LED (B12).
     let mut led = gpiob.pb12.into_push_pull_output(&mut gpiob.crh);
 
-    // Set up serial communication on pins A9 (Tx) and A10 (Rx), with 115200 baud.
-    let usart_pins = (gpioa.pa9.into_alternate_push_pull(&mut gpioa.crh), gpioa.pa10);
-    let mut protocol = SerialProtocol::new(dp.USART1, usart_pins, &mut afio, &mut rcc.apb2, clocks);
+    // Set up USB communications
+    let usb_pin_d_plus = gpioa.pa12;
+    let usb_pin_d_minus = gpioa.pa11;
+
+    let usb = Peripheral { usb: dp.USB, pin_dm: usb_pin_d_minus, pin_dp: usb_pin_d_plus };
+
+    let usb_bus = UsbBus::new(usb);
+    let serial = SerialPort::new(&usb_bus);
+
+    let usb_dev = UsbDeviceBuilder::new(&usb_bus, UsbVidPid(0x16c0, 0x27dd))
```

Comment:
I see we've chosen to be Van Ooijen Technische Informatica, CDC-ACM class devices (modems)

http://www.linux-usb.org/usb.ids

Any method to this?

Reply:
Ha to be honest this is from the USB serial example in the `stm32f1xx-hal`.

Since we're not distributing this as a product I'm not sure it matters much, but if there's a more appropriate VID/PID then I'm all for it.